### PR TITLE
Colormap docs

### LINKED ISSUE #175: Add colormap documentation
It should be straightforward to add a script porting the Matplotlib docs [from here](https://matplotlib.org/3.1.1/gallery/color/colormap_reference.html) to include the custom colormaps. 

```py
import numpy as np
import matplotlib.pyplot as plt


cmaps = [('Perceptually Uniform Sequential', [
            'viridis', 'plasma', 'inferno', 'magma', 'cividis']),
         ('Sequential', [
            'Greys', 'Purples', 'Blues', 'Greens', 'Oranges', 'Reds',
            'YlOrBr', 'YlOrRd', 'OrRd', 'PuRd', 'RdPu', 'BuPu',
            'GnBu', 'PuBu', 'YlGnBu', 'PuBuGn', 'BuGn', 'YlGn']),
         ('Sequential (2)', [
            'binary', 'gist_yarg', 'gist_gray', 'gray', 'bone', 'pink',
            'spring', 'summer', 'autumn', 'winter', 'cool', 'Wistia',
            'hot', 'afmhot', 'gist_heat', 'copper']),
         ('Diverging', [
            'PiYG', 'PRGn', 'BrBG', 'PuOr', 'RdGy', 'RdBu',
            'RdYlBu', 'RdYlGn', 'Spectral', 'coolwarm', 'bwr', 'seismic']),
         ('Cyclic', ['twilight', 'twilight_shifted', 'hsv']),
         ('Qualitative', [
            'Pastel1', 'Pastel2', 'Paired', 'Accent',
            'Dark2', 'Set1', 'Set2', 'Set3',
            'tab10', 'tab20', 'tab20b', 'tab20c']),
         ('Miscellaneous', [
            'flag', 'prism', 'ocean', 'gist_earth', 'terrain', 'gist_stern',
            'gnuplot', 'gnuplot2', 'CMRmap', 'cubehelix', 'brg',
            'gist_rainbow', 'rainbow', 'jet', 'nipy_spectral', 'gist_ncar'])]


gradient = np.linspace(0, 1, 256)
gradient = np.vstack((gradient, gradient))


def plot_color_gradients(cmap_category, cmap_list):
    # Create figure and adjust figure height to number of colormaps
    nrows = len(cmap_list)
    figh = 0.35 + 0.15 + (nrows + (nrows-1)*0.1)*0.22
    fig, axes = plt.subplots(nrows=nrows, figsize=(6.4, figh))
    fig.subplots_adjust(top=1-.35/figh, bottom=.15/figh, left=0.2, right=0.99)

    axes[0].set_title(cmap_category + ' colormaps', fontsize=14)

    for ax, name in zip(axes, cmap_list):
        ax.imshow(gradient, aspect='auto', cmap=plt.get_cmap(name))
        ax.text(-.01, .5, name, va='center', ha='right', fontsize=10,
                transform=ax.transAxes)

    # Turn off *all* ticks & spines, not just the ones with colormaps.
    for ax in axes:
        ax.set_axis_off()


for cmap_category, cmap_list in cmaps:
    plot_color_gradients(cmap_category, cmap_list)

plt.show()
```

### PR DESCRIPTION
**Pull request by kylebarron**
Add images describing colormaps. The README doesn't render correctly in the git diff, but it does on my fork: https://github.com/kylebarron/rio-tiler/blob/colormap-docs/docs/colormap.md

I think some of the images are easier to read when capitalization is kept, e.g. `OrRd` is clearly orange-red, whereas `orrd` is less clear. However since Linux uses a case-sensitive file-system, a user passing `OrRd` would fail, so I also coerce the name to lower case.

Closes #175

### CODE REVIEW

**Comment by kylebarron** [comment]
Are there only three new colormaps not from matplotlib?

**Comment by vincentsarago** [comment]
> Are there only three new colormaps not from matplotlib?

I think yes

**Comment by vincentsarago** [comment]
@kylebarron This is awesome. 

I wonder if we should add more info about the colormap itself. Meaning, we should maybe says the all are 0 -> 256, RGBA values ?

**Comment by kylebarron** [comment]
There could be a custom colormap section, but unless people are making custom colormaps, they don't need to know what a colormap object is

File: docs/colormap.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```py
+from rio_tiler.colormap import get_colormap
+from rio_tiler.utils import render
+
+colormap = get_colormap('cfastie')
+render(tile, mask, colormap=colormap)
+```
+
+![](img/geographic.png)
```
Comment (vincentsarago) [inline docs/colormap.md:17]: @kylebarron sorry for the delay, could we rename `geographic` to `custom` ?

**Pull request closed without merging**